src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1991,6 +1993,14 @@ private void connectHelper(ServerPortPlaceHolder serverInfo,
             connectionlogger.fine(toString() + " Connecting with server: " + serverInfo.getServerName() + " port: " + serverInfo.getPortNumber()
                     + " Timeout slice: " + timeOutsliceInMillis + " Timeout Full: " + timeOutFullInSeconds);
         }
+        
+        // Before opening the TDSChannel, calculate local hostname
+        // as the InetAddress.getLocalHost() takes more than usual time in certian OS and JVM combination, it avoids connection loss
```

Comment:
Suggestion: I think it should be good if we mention that in Sierra getLocalHost takes lot of time.